src/colvarmodule.cpp

```diff
@@ -1092,17 +1084,18 @@ int colvarmodule::calc_scripted_forces()
 int colvarmodule::write_restart_file(std::string const &out_name)
 {
   cvm::log("Saving collective variables state to \""+out_name+"\".\n");
-  proxy->backup_file(out_name);
-  std::ostream *restart_out_os = proxy->output_stream(out_name);
-  if (!restart_out_os) return cvm::get_error();
-  if (!write_restart(*restart_out_os)) {
+  std::ostream &restart_out_os = proxy->output_stream(out_name, "state file");
+  if (restart_out_os.bad()) return COLVARS_FILE_ERROR;
+  if (!write_restart(restart_out_os)) {
     return cvm::error("Error: in writing restart file.\n", COLVARS_FILE_ERROR);
   }
   proxy->close_output_stream(out_name);
-  if (cv_traj_os != NULL) {
+
+  if (proxy->output_stream_exists(cv_traj_name)) {
     // Take the opportunity to flush colvars.traj
-    proxy->flush_output_stream(cv_traj_os);
+    proxy->flush_output_stream(cv_traj_name);
   }
```

Comment:
This could be done without testing if the flush_output_stream function was made more permissive, like close_output_stream, and ignored streams that aren't open. I'm not sure the test will be very effective at catching bugs, so I'd rather simplify the calling code.

Reply:
Done